In [9]:
import pandas as pd

historical_df = pd.read_csv("../data/raw/historical_data.csv")
historical_df.shape

(211224, 16)

#### This dataset contains 211,224 trading execution records with 16 attributes describing user-level trading behavior including order execution, size, pricing, and profitability.

In [10]:
historical_df.columns

Index(['Account', 'Coin', 'Execution Price', 'Size Tokens', 'Size USD', 'Side',
       'Timestamp IST', 'Start Position', 'Direction', 'Closed PnL',
       'Transaction Hash', 'Order ID', 'Crossed', 'Fee', 'Trade ID',
       'Timestamp'],
      dtype='object')

#### Each row represent a single executed trade event.
#### Primary entity appears to be: Account (user-level identifier)

#### Entity Mapping:
    - User ID: Account
    - Asset: Coin
    - Timestamp: Timestamp / Timestamp IST
    - Action: Side, Direction
    - Trade size: Size USD, Size Tokens
    - Outcome: Closed PnL
    - Market context: Execution Price
    - Transaction metadata: Order ID, Trade ID, Transaction Hash

#### Behavioral Features Definition

We define the following user-level behavioral features for anomaly detection:

1. trades_per_day
2. total_trade_volume_per_day
3. avg_trade_size
4. time_between_trades (optional / derived)


#### Feature Feasibility Check

We validate whether the dataset supports computation of the defined behavioral features.

All 4 core behavioral features are directly available from transaction-level fields in the dataset without extra assumptions.

#### Feature <-> Column Mapping

    - trades_per_day -> Account + Timestamp
    - total_trades_volume_per_day -> Size USD + Account + Timestamp
    - avg_trading_size -> Size USD aggregated per Account/Day
    - time_between_trades -> Timestamp deltas per Account

#### Feature 1: trades_per_day

In [12]:
historical_df["Timestamp"] = pd.to_datetime(historical_df['Timestamp'], unit='ms')
historical_df["date"] = historical_df["Timestamp"].dt.date
historical_df["date"]

0         2024-10-27
1         2024-10-27
2         2024-10-27
3         2024-10-27
4         2024-10-27
             ...    
211219    2025-06-15
211220    2025-06-15
211221    2025-06-15
211222    2025-06-15
211223    2025-06-15
Name: date, Length: 211224, dtype: object